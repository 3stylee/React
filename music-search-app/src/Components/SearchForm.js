import { Container, FormControlLabel, Switch, Box} from "@mui/material";
import SearchIcon from '@mui/icons-material/Search';
import React from "react";
import { SearchButton, SearchInput, SearchBar } from "./StyledComponents";

const SearchForm = props => (
    <Container>
        <SearchBar>
            <SearchInput 
                fullWidth 
                focused
                label="Search" 
                variant="outlined" 
                color="secondary"
                onChange={props.onSearchChanged}
            />
            <SearchButton
                variant="outlined" 
                color="secondary" 
                onClick={props.onClick}
            >
                <SearchIcon/>
            </SearchButton>
        </SearchBar>
        <Box>
            {Object.keys(props.filters).map(filterName => (
                <FormControlLabel 
                    control={<Switch defaultChecked name={filterName} color="secondary" onChange={props.onSwitchChanged}/>} 
                    label={filterName} 
                    key={filterName}
                />)
            )}
        </Box>
    </Container>
)

export default SearchForm;