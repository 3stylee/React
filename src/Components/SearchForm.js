import { Container, FormControlLabel, Switch, Box } from '@mui/material';
import SearchIcon from '@mui/icons-material/Search';
import React from 'react';
import { SearchButton, SearchInput, SearchBar } from './StyledComponents';

const SearchForm = props => {
	return (
		<Container maxWidth={false} sx={{ paddingstart: '1rem' }}>
			<SearchBar>
				<SearchInput
					fullWidth
					focused
					placeholder="Enter an artist, song, or album title..."
					label="Search"
					variant="outlined"
					color="secondary"
					onChange={props.onSearchChanged}
				/>
				<SearchButton
					variant="outlined"
					color="secondary"
					onClick={props.onClick}>
					<SearchIcon />
				</SearchButton>
			</SearchBar>
			<Box>
				{Object.keys(props.filters).map(filterName => (
					<FormControlLabel
						control={
							<Switch
								checked={props.filters[filterName]}
								name={filterName}
								color="secondary"
								onChange={props.onSwitchChanged}
							/>
						}
						label={filterName}
						key={filterName}
					/>
				))}
			</Box>
		</Container>
	);
};

export default SearchForm;
