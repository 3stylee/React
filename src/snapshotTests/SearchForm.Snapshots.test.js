import React from 'react';
import renderer from 'react-test-renderer';
import SearchForm from '../Components/SearchForm';

it('should set all filter switches to on if the filters are all set to true', () => {
	const tree = renderer.create(
		<SearchForm
			onClick={jest.fn()}
			filters={{
				Songs: true,
				Artists: true,
				Albums: true,
			}}
			onSearchChanged={jest.fn()}
			onSwitchChanged={jest.fn()}
		/>
	);

	expect(tree).toMatchSnapshot();
});
