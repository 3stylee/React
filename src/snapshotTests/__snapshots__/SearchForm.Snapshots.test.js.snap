// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set all filter switches to on if the filters are all set to true 1`] = `
<div
  className="MuiContainer-root css-12eofg2-MuiContainer-root"
>
  <div
    className="MuiBox-root css-1maxgwy"
  >
    <div
      className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-14xrz3f-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorSecondary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-179z3gj-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={true}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Search
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-1j0w0ul-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter an artist, song, or album title..."
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-14lo706"
          >
            <span>
              Search
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-167o6lf-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="SearchIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </button>
  </div>
  <div
    className="MuiBox-root css-0"
  >
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked Mui-checked css-priqlz-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            name="Songs"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Songs
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked Mui-checked css-priqlz-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            name="Artists"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Artists
      </span>
    </label>
    <label
      className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
    >
      <span
        className="MuiSwitch-root MuiSwitch-sizeMedium css-julti5-MuiSwitch-root"
      >
        <span
          className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary Mui-checked Mui-checked css-priqlz-MuiButtonBase-root-MuiSwitch-switchBase"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            checked={true}
            className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
            name="Albums"
            onChange={[Function]}
            required={false}
            type="checkbox"
          />
          <span
            className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
          />
        </span>
        <span
          className="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
        />
      </span>
      <span
        className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
      >
        Albums
      </span>
    </label>
  </div>
</div>
`;
